config/core/resources/cloudauditlogssource.yaml

```diff
@@ -36,158 +36,297 @@ spec:
+                    - type
+                    - status
+              serviceAccountName:
+                type: string
+              sinkUri:
+                type: string
+              ceAttributes:
+                type: array
+                items:
                   type: object
-                  description: >
-                    Extensions specify what attribute are added or overridden on the outbound event. Each
-                    `Extensions` key-value pair are set on the event as an attribute extension independently.
-                  x-kubernetes-preserve-unknown-fields: true
-            serviceAccountName:
-              type: string
-              description: >
-                Kubernetes service account used to bind to a google service account to poll the Cloud Pub/Sub Subscription.
-                The value of the Kubernetes service account must be a valid DNS subdomain name.
-                (see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names)
-            secret:
-              type: object
-              description: >
-                Credential used to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the
-                Subscription, only to poll it. The value of the secret entry must be a service account key in
-                the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys).
-                Defaults to secret.name of 'google-cloud-key' and secret.key of 'key.json'.
-              properties:
-                name:
-                  type: string
-                key:
-                  type: string
-                optional:
-                  type: boolean
-            project:
-              type: string
-              description: >
-                Google Cloud Project ID of the project into which the topic should be created. If omitted uses
-                the Project ID from the GKE cluster metadata service.
-            serviceName:
-              type: string
-            methodName:
-              type: string
-            resourceName:
-              type: string
-        status:
-          type: object
-          properties:
-            observedGeneration:
-              type: integer
-              format: int64
-            conditions:
-              type: array
-              items:
+                  properties:
+                    type:
+                      type: string
+                    source:
+                      type: string
+              projectId:
+                type: string
+              topicId:
+                type: string
+              subscriptionId:
+                type: string
+              stackdriverSink:
+                type: string
+                description: >
+                  ID of the Stackdriver sink used to publish audit log messages.
+  - << : *version
+    name: v1beta1
+    served: true
+    storage: true
+  - <<: *version
```

Comment:
The yaml looks kind of weird.
As I suggested, you use yaml merge https://yaml.org/type/merge.html to try to deduplicate common fields.
But you still copy all fields under `v1` version, in this case, what's the usage of using yaml merge https://yaml.org/type/merge.html.
I think you should combine all duplicate methods with yaml merge for reuse.

Reply:
Similar issues for other resoureces

Reply:
I'm not an expert of yaml. @Harwayne  Any ideas?

Reply:
This should be fixed.

Reply:
This looks correct to me. My memory is that merge is an optional part of YAML, but it looks like Kubernetes supports it, so we can use it.

Reply:
The yaml merge syntax is correct.  But previously we duplicated all fields. I think currently it is correct